### PR TITLE
Add version modal to protocol show page [SCI-7985]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-7985](https://scinote.atlassian.net/browse/SCI-7985)

### What was done
Add version modal to protocol show page

[SCI-7985]: https://scinote.atlassian.net/browse/SCI-7985?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/vue/protocol/protocolMetadata.vue [hunk 1]
```diff
@@ -113,6 +113,21 @@
             this.$emit('update', result.data.attributes)
           }
         });
+      },
+      openVersionsModal() {
+        console.log(this.protocol.attributes.urls.versions_modal_url)
```
Comment (rekonder) [inline app/javascript/vue/protocol/protocolMetadata.vue:118]: Remove this

File: app/javascript/packs/vue/protocol.js [hunk 2]
```diff
@@ -6,11 +6,21 @@ import ProtocolContainer from '../../vue/protocol/container.vue';
...
 Vue.use(TurbolinksAdapter);
 Vue.prototype.i18n = window.I18n;
+Vue.prototype.inlineEditing = window.inlineEditing;
 Vue.prototype.ActiveStoragePreviews = window.ActiveStoragePreviews;
 
 window.initProtocolComponent = () => {
   Vue.prototype.dateFormat = $('#protocolContainer').data('date-format');
 
+  $('.protocols-show').on('click', '#protocol-versions-modal .delete-draft', (e) => {
+    console.log(e)
```
Comment (rekonder) [inline app/javascript/packs/vue/protocol.js:16]: Solve this

**Review by rekonder — CHANGES_REQUESTED** [review]

**Review by rekonder — APPROVED** [review]

File: app/controllers/protocols_controller.rb [hunk 3]
```diff
@@ -196,8 +196,11 @@ def destroy_draft
       log_activity(:protocol_template_draft_deleted,
                    nil,
                    protocol: @protocol.id)
+      redirect_to protocols_path
+    rescue ActiveRecord::RecordNotDestroyed => e
+      Rails.logger.error e.message
```
Comment (okriuchykhin) [inline app/controllers/protocols_controller.rb:201]: you forgot to rollback transaction

**Review by okriuchykhin — CHANGES_REQUESTED** [review]
Check comments

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**